Overlay for Microchip MCP23S08/17 SPI gpio expanders

wavelet2 opened the pull request.

Commit:
Added Overlay for Microchip MCP23S08/17 SPI gpio expanders

Commit:
Added Overlay for Microchip MCP23S08/17 SPI gpio expanders

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -622,6 +622,30 @@ Params: gpiopin                 Gpio pin connected to the INTA output of the
+        If interrupts are enabled for a device on a given CS# on a SPI bus, that
+        device must be the only one present on that SPI bus/CS#.
+Load:   dtoverlay=mcp23s17,<param>=<val>
+Params: mcp23s08-spi<n>-<m>-present  4-bit interger, bitmap indicating MCP23S08
```
`integer` is the correct spelling.

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -622,6 +622,30 @@ Params: gpiopin                 Gpio pin connected to the INTA output of the
+Params: mcp23s08-spi<n>-<m>-present  4-bit interger, bitmap indicating MCP23S08
+                                     devices present on SPI<n>, CS#<m>
+
+        mcp23s17-spi<n>-<m>-present  8-bit interger, bitmap indicating MCP23S17
```
These identifiers are very long, and that has caused you to change the indentation on the descriptions, which is not allowed. You might want to consider shortening some of the names, but if not then I suggest you start the descriptions on the following lines.

pelwell commented:
In addition to fixing the formatting issues you will also need to add a line to .../overlays/Makefile, otherwise the overlay will not be compiled.

pelwell commented:
I also think that the support for both underscores and hyphens doesn't serve any useful purpose, and it makes the file even more unwieldy; I would suggest standardizing on one (most overlays have used underscores). Whichever you settle on, make sure the README agrees.

Commit:
Update mcp23s17-overlay.dts

Fixed spelling error.  Removed underscore option for dtparams.

Commit:
Update README

Corrected spelling of 'integer'.  Shortened dtparam names to comply with indenting rules.

Commit:
Update mcp23s17-overlay.dts

Shortened dtparam names.

Commit:
Update Makefile

Added target for MCP23S08/17 overlay.

wavelet2 commented:
pelwell, I have updated the files as you suggest and added a target to the Makefile.  Regards, wavelet2.

pelwell commented:
Congratulations - you win the prize for the most complicated overlay yet. Your changes look good - nice job.

pelwell commented:
Previously I would have asked you to squash your commits into one, but instead I used the new squash-and-merge feature to do it myself. This way works better because it preserves the history in the PR.
